nengo/spa/pointer.py

```diff
@@ -150,35 +139,24 @@ def distance(self, other):
         """
         return 1 - self.compare(other)
 
-    def __invert__(self):
-        """Return a reorganized vector that acts as an inverse for convolution.
-
-        This reorganization turns circular convolution into circular
-        correlation, meaning that ``A*B*~B`` is approximately ``A``.
+    def mse(self, other):
+        """Return the mean-squared-error between two vectors."""
+        return np.sum((self - other).v**2) / len(self.v)
 
-        For the vector ``[1, 2, 3, 4, 5]``, the inverse is ``[1, 5, 4, 3, 2]``.
-        """
-        return SemanticPointer(data=self.v[-np.arange(len(self))])
 
-    def __len__(self):
-        """Return the number of dimensions in the vector."""
-        return len(self.v)
+class Identity(SemanticPointer):
+    def __init__(self, n_dimensions):
+        data = np.zeros(n_dimensions)
+        data[0] = 1.
+        super(Identity, self).__init__(data)
 
-    def copy(self):
-        """Return another semantic pointer with the same data."""
-        return SemanticPointer(data=self.v)
 
-    def mse(self, other):
-        """Return the mean-squared-error between two vectors."""
-        return np.sum((self - other).v**2) / len(self.v)
+class AbsorbingElement(SemanticPointer):
```

Comment:
I think for our purposes it pretty much is. You get a scaled version only if the other vector is not unit length and every information about what the vector except its length will be lost.